File: Java/q1047.java
// Tempo de Jogo com Minutos

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
 
       Scanner input = new Scanner(System.in);
       int h1 = input.nextInt();
       int m1 = input.nextInt();
       int h2 = input.nextInt();
       int m2 = input.nextInt();
       
       int total = (h2*60 + m2) - (h1*60 + m1);
       
       if (total < 0){
           total += 1440; 
       }
    
       if (total == 0) {
           System.out.println("O JOGO DUROU 24 HORA(S) E 0 MINUTO(S)");
       } else {
           int horatotal = total / 60;
           int minutototal = total % 60;
           
           System.out.printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)%n", horatotal, minutototal);
       }
    }
}

File: Cpp/q1047.cpp
#include <iostream>
using namespace std;

int main(){

  int h1, m1, h2, m2, horatotal, minutototal;
  cin >> h1 >> m1 >> h2 >> m2;

  horatotal = h2 - h1;
  minutototal = m2 - m1;

    if (minutototal < 0){
        horatotal--;
        minutototal += 60;
    }

    if (horatotal < 0){
        horatotal += 24;
    }
  
    if (horatotal == 0 && minutototal == 0){
        horatotal += 24;
    }

    cout << "O JOGO DUROU " << horatotal << " HORA(S) E " << minutototal << " MINUTO(S)" << endl;

    return 0;
}
    